### PR TITLE
CN-4258 apply for loan flow improvements

### PR DESCRIPTION
**Pull request by ShamanShaulic**
PR Checklist:
- [x] I have named the PR correctly
- [x] I have added at least two reviewers
- [x] I have assigned the PR to myself
- [x] I have merged develop into my feature branch and resolved conflicts
- [x] I have tested all new features locally
- [x] I have tested relevant flow locally

### How to name a PR
- If it includes only one issue [CN-1234 Issue name]
- If it includes many issues [CN-1234, CN-2345, CN-3456 What was updated/changed/added]
- If the PR is not ready yet, add [DRAFT] in front of the name

### How to test locally
1. Test happy path
2. Test other paths
3. Remove all warnings

### How to relevant flow locally
- If working on part of CelPay, check the whole happy path

### CODE REVIEW

File: app/components/screens/BorrowChooseLoan/BorrowChooseLoan.js [hunk 1]
```diff
@@ -40,14 +40,20 @@ class BorrowChooseLoan extends Component {
         cardCopy: "Take out a cash loan against your crypto.",
         lightImage: require("../../../../assets/images/illustration-borrow-dollars.png"),
         darkImage: require("../../../../assets/images/illustration-borrow-dollars.png"),
-        onPressAction: () => actions.navigateTo("LoanPaymentCoin"),
+        onPressAction: () => {
+          actions.navigateTo("BorrowEnterAmount");
+          actions.updateFormField("coin", "USD");
```
Comment (fjovakaric) [inline app/components/screens/BorrowChooseLoan/BorrowChooseLoan.js]: should also set loan type here

File: app/components/screens/BorrowChooseLoan/BorrowChooseLoan.js [hunk 2]
```diff
@@ -40,14 +40,20 @@ class BorrowChooseLoan extends Component {
...
       },
       {
         cardTitle: "Borrow Stablecoins",
         cardCopy: "Choose a stable coin to take your loan in.",
         lightImage: require("../../../../assets/images/illustration-borrow-stablecoins.png"),
         darkImage: require("../../../../assets/images/illustration-borrow-stablecoins.png"),
-        onPressAction: () => actions.navigateTo("LoanPaymentCoin"),
+        onPressAction: () => {
+          actions.navigateTo("BorrowEnterAmount");
+          actions.updateFormField("coin", "");
```
Comment (fjovakaric) [inline app/components/screens/BorrowChooseLoan/BorrowChooseLoan.js]: should also set loan type here

File: app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js [hunk 3]
```diff
@@ -201,29 +205,27 @@ class BorrowEnterAmount extends Component {
...
-                coinCompliance={coinSelectItems}
-                navigateTo={actions.navigateTo}
-              />
-            </View>
+            <CoinPicker
+              type={"withIcon"}
+              onChange={(field, value) =>
+                actions.updateFormFields({
+                  [field]: value,
+                  loanType:
```
Comment (fjovakaric) [inline app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js]: this should be set on previous screen

File: app/components/molecules/CoinPicker/CoinPicker.js [hunk 4]
```diff
@@ -8,136 +8,122 @@ import Icon from "../../atoms/Icon/Icon";
...
   static propTypes = {
-    coinList: PropTypes.instanceOf(Array),
     updateFormField: PropTypes.func.isRequired,
     field: PropTypes.string.isRequired,
-    value: PropTypes.string,
-    onCoinSelect: PropTypes.func,
+    coin: PropTypes.string,
     defaultSelected: PropTypes.string,
-    coinCompliance: PropTypes.instanceOf(Array).isRequired, // coinComplience are coins which are eligable to show up depending on screen on which they came from are opened
+    coinCompliance: PropTypes.instanceOf(Array).isRequired,
```
Comment (fjovakaric) [inline app/components/molecules/CoinPicker/CoinPicker.js]: rename to `availableCoins` or `coins`

File: app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js [hunk 5]
```diff
@@ -58,7 +59,7 @@ class BorrowEnterAmount extends Component {
       .filter(c => loanCompliance.loan_coins.includes(c.short))
       .map(c => ({ label: `${c.displayName}  (${c.short})`, value: c.short }));
 
-    coinSelectItems.push({ label: `US Dollar ($)`, value: "USD" });
+    coinSelectItems.push({ label: `Dollar (USD)`, value: "USD" });
```
Comment (fjovakaric) [inline app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js]: Do we need this?

Comment (ShamanShaulic) [reply app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js]: guess not

**Pull request merged**